kustomize/resource_kustomization.go

```diff
@@ -177,10 +208,49 @@ func kustomizationResourceRead(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
 
+func kustomizationResourceRead(d *schema.ResourceData, m interface{}) error {
+	kubeconfigPath := d.Get("kubeconfig_path").(string)
+	kubeconfigRaw := d.Get("kubeconfig_raw").(string)
+	kubeContext := d.Get("context").(string)
+
+	client := m.(*Config).Client
+	mapper := m.(*Config).Mapper
+
+	if kubeconfigPath != "" || kubeconfigRaw != "" || kubeContext != "" {
+		k, err := initializeClient(kubeconfigRaw, kubeconfigPath, kubeContext, false, 1)
+
+		if k == nil {
+			return fmt.Errorf("provider kustomization: %s", err)
```

Comment:
```suggestion
			return fmt.Errorf("failed to initialize client: %w", err)
```

Reply:
this preserves the error message from the master branch here:

https://github.com/kbst/terraform-provider-kustomization/blob/830ec83394a2e1d1182b4212389f3fe00fb7c187/kustomize/provider.go#L136-L145